### DISCUSSION TITLE
SystemUser for Silent Messages

### DISCUSSION THREAD

**Discussion by rupinderjeet** [GetStream/stream-chat-android#1683] [Q&A] [answered]
Documentation for silent messages mentions a `systemUser` user instance to use when sending these messages.

If I pass a blank `User()` instance to it, the message is sent from current user.

I thought I could use this to show stuff like "User X added User Y" in the center. Is it not for this?

https://getstream.io/chat/docs/android/silent_messages/

**Comment by leandroBorgesFerreira** [comment]
Silent messages are just messages that don't mark the channel as unread and don't increase the count of unread messages. 
If I understand correct you would like a custom type of message that you can use to notify users about some update in the channel. I believe this would be the case for a custom message.

**Comment by rupinderjeet** [comment]
Ok, thank you.

About the custom messages like this, I do not want the message to be associated with any of channel members. Will the sender user need to be someone with admin/moderator permission so that sender can send such custom messages to all the channels?

Perhaps, this user need to be maintained by a backend of some sort which monitors events happening in channels and posts these messages.

Is this right?

**Comment by leandroBorgesFerreira** [comment]
You'll need to send a system message, take a look at the message types: https://getstream.io/chat/docs/other-rest/message_format/?language=#message-types.

**Comment by zsmb13** [comment, marked as answer]
There are two ways of achieving this:
- Create a dedicated user for this server-side (call it a system user), and send silent messages with that user when needed. These will have the type `regular`, but can be rendered differently by adding custom data to them.
- Use the workaround of issuing channel updates, as described here https://github.com/GetStream/stream-chat-js/issues/618#issuecomment-786855205
